risc0/zkvm/prove/method_id.cpp

```diff
@@ -66,4 +66,18 @@ void writeMethodID(const std::string& filename, const MethodID& id) {
   }
 }
 
+namespace rust {
+
+MethodID::MethodID(const std::string& elf_path) : id(makeMethodID(elf_path)) {}
+
+std::unique_ptr<MethodID> new_method_id(const std::string& elf_path) {
+  return std::make_unique<MethodID>(elf_path);
+}
+
+void MethodID::write(const std::string& filename) const {
```

Comment:
I'm not really excited about the API requiring all this stuff to be in OS files, but I guess that's how it is currently, and it's relatively straightforward to change later.

Reply:
This PR was getting pretty big so decided to avoid doing functional changes for now, but I agree.